### PR TITLE
141 sort properties

### PR DESCRIPTION
**Pull request by raduiordache**
@jberends - Can you also review this, bitte schon?

### CODE REVIEW

File: pykechain/models/part.py [hunk 1]
```diff
@@ -492,3 +492,27 @@ def as_dict(self):
...
+
+        Example
+        -------
+        >>> front_fork = client.scope('Bike Project').model('Front Fork')
+        >>> front_fork.order_properties(['Material', 'Height (mm)', 'Color'])
+
+        """
+        assert self.category == Category.MODEL
+
+        if isinstance(list_property_names, list):
```
Comment (jberends) [inline pykechain/models/part.py]: only names, or also property objects from pykechain.

Throughout the code we use mostly both. I agree that names should be 'unique', but there is no guarentee that these are. In that sense the ordering should also be done by a list of property (model) id's

File: pykechain/models/part.py [hunk 2]
```diff
@@ -492,3 +492,27 @@ def as_dict(self):
...
+            order_dict = dict()
+
+            for prop in list_property_names:
+                order_dict[self.property(name=prop).id] = list_property_names.index(prop)
```
Comment (jberends) [inline pykechain/models/part.py]: in continuation of my comment above, the transformation from names to id's is already done here as the API requires the id's indeed. 

Please make it id and name agnostic

File: pykechain/models/part.py [hunk 3]
```diff
@@ -492,3 +492,27 @@ def as_dict(self):
...
+
+            r = self._client._request('PUT', self._client._build_url('part', part_id=self.id),
```
Comment (jberends) [inline pykechain/models/part.py]: check response code here pls

**Comment by raduiordache** [comment]
@jberends Changes implemented, please review them :smile:

File: pykechain/models/part.py [hunk 4]
```diff
@@ -492,3 +494,43 @@ def as_dict(self):
...
+                                      data=dict(
+                                          property_order=json.dumps(order_dict)
+                                      ))
+            if r.status_code != requests.codes.ok:  # pragma: no cover
+                raise APIError("Could not reorder properties")
+
+        else:
+            raise TypeError('Expected a list of strings or Property() objects, got a {} object'.
```
Comment (jberends) [inline pykechain/models/part.py]: why not using an assert on L517? instead of a 'wide' if statement?

File: pykechain/models/part.py [hunk 5]
```diff
@@ -492,3 +494,43 @@ def as_dict(self):
...
+        >>> front_fork.order_properties([material, height, color])
+
+
+        """
+        assert self.category == Category.MODEL
+
+        if isinstance(property_list, list):
+            order_dict = dict()
+
+            for prop in property_list:
```
Comment (jberends) [inline pykechain/models/part.py]: I like 👍

**Pull request merged**